src/wp-includes/class-wp-theme-json.php

```diff
@@ -1934,6 +1934,16 @@ protected static function compute_style_properties( $styles, $settings = array()
 			return $declarations;
 		}
 
+		$can_use_cached = ! wp_is_development_mode( 'theme' );
```

Comment:
As far as I can tell, we may be able to avoid the `wp_is_development_mode()` check here entirely. Here's why:
* The cache key depends on the given method arguments. This means it'll always be "fresh". For example, if a style changes, a new cache key will be used.
* The only exception is the call to `wp_get_typography_font_size_value()` below, which relies on other data (e.g. `wp_get_global_settings()`). So maybe we can cache everything except that one? Would that still be beneficial for performance?

That would mean that we cache the result of the foreach loop, except for the `font-size` CSS property we simply store the `$value` without calling `wp_get_typography_font_size_value()`.

After that loop, even if we got the result from cache, we can cycle through the loop of the results and call `wp_get_typography_font_size_value()` for any entry that is using the `font-size` CSS property.

Reply:
@felixarntz 

I agree that the check for the development mode here doesn't bring any value. For that reason, I removed it.

As for your concern around `wp_get_global_settings()` for the case when users tweak values, I think that we can get away with making that part of the recipe for the cache key. Here's [how](https://github.com/WordPress/wordpress-develop/pull/5567/commits/d707a79b5eb8d9ba81fb808e234138d2ead8264a).

What do you think?